### PR TITLE
session 9

### PR DESCRIPTION
**Pull request by Dragos-U**

### CODE REVIEW

File: src/main/java/org/session9/homework/collegemanagementsystem/Course.java [hunk 1]
```diff
@@ -0,0 +1,61 @@
+package org.session9.homework.collegemanagementsystem;
+
+public class Course {
+
+    private String schedule;
```
Comment (AlexandruIliescu) [inline src/main/java/org/session9/homework/collegemanagementsystem/Course.java:5]: For easier maintenance, we can use an object that handles time. Even though we need in this situation a day and a time, we can maybe combine them into a single object (e.g. for the day we can build an `enum` with all days of the week, or we can use a String, and for the time of the course, LocalTime)

File: src/main/java/org/session9/homework/collegemanagementsystem/Professor.java [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+import java.time.LocalDate;
+import java.util.ArrayList;
+import java.util.List;
+
+public class Professor {
+
+    private String firstName;
+    private String lastName;
+    private char sex;
+    private short age;
```
Comment (AlexandruIliescu) [inline src/main/java/org/session9/homework/collegemanagementsystem/Professor.java]: If we have `dateOfBirth`, do we need `age` property? In our case, having `dateOfBirth` unlocks the possibility to compute dynamically the age, anytime we want.

File: src/main/java/org/session9/homework/collegemanagementsystem/Professor.java [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+    private LocalDate dateOfBirth;
+    private String CNP;
```
Comment (AlexandruIliescu) [inline src/main/java/org/session9/homework/collegemanagementsystem/Professor.java:13]: As a rule of thumb, we should try to use only properties in English, like `uniqueId` or something similar, for this case.

File: src/main/java/org/session9/homework/fitnesstrackingsystem/User.java [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+        System.out.println("\nTotal distance : " + getTotalDistance() + " km");
+        System.out.println("Total duration : " + getTotalDuration() + " min");
+        System.out.println("Total lifted weight :" + getTotalLiftedWeigth() + " kg");
+    }
+
+    public void checkGoals() {
+        for (String workoutType : weeklyGoals.keySet()) {
+            double goal = weeklyGoals.get(workoutType);
+            String achievement = null;
+            if (workoutType.equals("running")) {
```
Comment (AlexandruIliescu) [inline src/main/java/org/session9/homework/fitnesstrackingsystem/User.java]: We can use a switch maybe in this case? Having a switch will make our code easier to scale in time. What if our client wants to add 3 workout types? 
Also, `running`, `yoga`, and `weightLifting` can be declared as constants and we can maybe make `equalsIgnoreCase` compare for this particular situation. We should not care if it's `yoga` or `Yoga`, the result of the comparison should be true in both scenarios.

File: src/main/java/org/session9/homework/schoolgradebook/CourseGradeBook.java [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+        }
+        System.out.println("\nThe average grade per course is: " + gradeSum / students.size());
+    }
+
+    public void printStudentGPA(){
+        for(Student student : students){
+            System.out.println(student.getFullName() + " GPA is : " + student.getGradePointAverage());
+        }
+    }
+
```
Comment (AlexandruIliescu) [inline src/main/java/org/session9/homework/schoolgradebook/CourseGradeBook.java:31]: Please don't forget to remove empty lines and to do the code formatting.

File: src/main/java/org/session9/homework/schoolgradebook/SchoolGradebookManagement.java [hunk 6]
```diff
@@ -0,0 +1,33 @@
+package org.session9.homework.schoolgradebook;
+
+public class SchoolGradebookManagement {
+
+    public static void main(String[] args) {
+
```
Comment (AlexandruIliescu) [inline src/main/java/org/session9/homework/schoolgradebook/SchoolGradebookManagement.java]: Please remove empty lines like this, as we agreed on our [Coding guidelines](https://docs.google.com/document/d/1YT_rjqUlZ8cn8ZI7QAD52LXZq8oVD5_Gq0b5uO40fYk/edit?usp=sharing) manual.

**Review by AlexandruIliescu — APPROVED** [review]

File: src/main/java/org/session9/homework/collegemanagementsystem/Professor.java (hunk 3 above)
Comment (Dragos-U) [reply src/main/java/org/session9/homework/collegemanagementsystem/Professor.java:13]: agree. However, CNP was given as a requirement in the problem, and I was sticking to that. The challenge should have contained some other variable reference name.

Comment (AlexIliescu11) [reply src/main/java/org/session9/homework/collegemanagementsystem/Professor.java:13]: You can have a Romanian or Pakistan client that will send you requirements in their native language, this doesn't make any difference. We know that we should think/code our product in English so it's up to us how we do the design of classes, methods and properties.

**Pull request closed without merging**